base/sparse/abstractsparse.jl

```diff
@@ -21,3 +23,22 @@ issparse{T, A<:AbstractSparseMatrix}(S::UpperTriangular{T, A}) = true
 issparse{T, A<:AbstractSparseMatrix}(S::LinAlg.UnitUpperTriangular{T, A}) = true
 
 indtype{Tv,Ti}(S::AbstractSparseArray{Tv,Ti}) = Ti
+
+function union!{Tv,Ti,N}(s::AbstractSet, x::AbstractSparseArray{Tv,Ti,N})
+    vals = nonzeros(x)
+    length(vals) == length(x) ? union!(s, vals) : union!(s, [zero(Tv); vals])
+end
+
+# assumes that AbstractSparseArray type stores nonzeros in column major order
+# this assumption is subject to change in the future
+function unique{Tv,Ti,N}(x::AbstractSparseArray{Tv,Ti,N})
+    vals = nonzeros(x)
+    if length(x) == length(vals)
+        return unique(vals)
+    else
+        # find the first zero valued element
+        i = findfirst(x, zero(Tv))
```

Comment:
> this seems potentially really inefficient

Yes. I was optimize for what I was imagining was the typical case of random sparsity.

>  oh, and the index i with respect to x is not correct to then use as an index into vals.

I probably was thinking that `vals` would not contain zeros when I wrote this. I think the fixup @nalimilan suggest would be good.